app/src/pages/p/[projectSlug]/billing/invoices/[id].tsx

```diff
@@ -0,0 +1,149 @@
+import { useInvoice, useSelectedProject } from "~/utils/hooks";
+
+export default function BillingTabs() {
+  const selectedProject = useSelectedProject().data;
+
+  const router = useRouter();
+  const id = router.query.id as string;
+
+  const { data } = useInvoice(id, 10000);
+  const invoice = data?.invoice;
+
+  if (!invoice) return <></>;
+
+  // Checks if the description is an array of objects with string values to prevent rendering errors
+  const isCorrectDescriptionFormat = (value: unknown): value is { [key: string]: string }[] => {
+    return Array.isArray(value) && value.every((item) => item && typeof item === "object");
+  };
+
+  return (
+    <AppShell title="Billing" requireAuth>
+      <VStack position="sticky" left={0} right={0} w="full">
+        <PageHeaderContainer>
+          <Breadcrumb>
+            <BreadcrumbItem>
+              <ProjectBreadcrumbContents projectName={selectedProject?.name} />
+            </BreadcrumbItem>
+            <BreadcrumbItem>
+              <ProjectLink href="/billing">
+                <Flex alignItems="center" _hover={{ textDecoration: "underline" }}>
+                  <Icon as={LiaFileInvoiceDollarSolid} boxSize={4} mr={2} /> Billing
+                </Flex>
+              </ProjectLink>
+            </BreadcrumbItem>
+            <BreadcrumbItem isCurrentPage>
+              <Text>{invoice?.slug}</Text>
+            </BreadcrumbItem>
+          </Breadcrumb>
+        </PageHeaderContainer>
+
+        <VStack px={8} py={8} alignItems="flex-start" spacing={4} w="full">
+          <Text fontSize="2xl" fontWeight="bold">
+            Invoice {invoice?.slug}
+          </Text>
+          <Card width="100%" overflowX="auto">
+            <VStack py={8}>
+              <HStack px={8} w="900px" justifyContent="space-between" paddingBottom={5}>
+                <VStack alignItems="start">
+                  <Text fontSize="2xl" fontWeight="bold">
+                    Invoice
+                  </Text>
+                  <Text fontSize="2xl" fontWeight="bold">
+                    OpenPipe
+                  </Text>
+                </VStack>
+                <VStack alignItems="end">
+                  <Text fontSize="2xl" fontWeight="bold" textAlign={"right"}>
+                    {invoice?.slug}
+                  </Text>
+                  <Text fontSize="2xl" fontWeight="bold" textAlign={"end"}>
+                    {invoice?.billingPeriod}
+                  </Text>
+                </VStack>
+              </HStack>
+
+              <TableContainer>
+                <Table size="lg" width="900px">
+                  <Thead>
+                    <Tr>
+                      <Th></Th>
+                      <Th>Description</Th>
+                      <Th isNumeric>Amount</Th>
+                    </Tr>
+                  </Thead>
+                  <Tbody>
+                    {isCorrectDescriptionFormat(invoice.description) &&
```

Comment:
I would probably move this display logic and its associated type checking to a separate component within the same file.

Reply:
In this case, I'm not sure; I personally like to have the whole table in a single component if it is not that big, or at least extract rows into separate components. But I can easily do it to see how it looks.